### PR TITLE
[AV-1842] Made prompts mockable

### PR DESCRIPTION
**Pull request by dasconnor**
I made the prompts package an interface and generated a mock prompt interface to go along with it. I then stored a prompt instance in the application object. Tests that want to go over code with prompts merely need to set a mock prompt object in `app`.

EDIT: I also added a script to regenerate mock objects. I renamed all of the generated files to match the output of the `mockery` script.

### CODE REVIEW

File: internal/mocks/setDefaultSnapshotFunc.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Code generated by mockery v2.14.0. DO NOT EDIT.
```
Comment (felipemadero) [inline internal/mocks/setDefaultSnapshotFunc.go:1]: is this used in tests?

File: pkg/prompts/prompts.go [hunk 2]
```diff
@@ -16,6 +16,24 @@ const (
...
+	CaptureAddress(promptStr string) (common.Address, error)
+	CaptureExistingFilepath(promptStr string) (string, error)
+	CaptureYesNo(promptStr string) (bool, error)
+	CaptureNoYes(promptStr string) (bool, error)
+	CaptureList(promptStr string, options []string) (string, error)
+	CaptureString(promptStr string) (string, error)
+	CaptureIndex(promptStr string, options []common.Address) (int, error)
+}
+
+type realPrompter struct{}
```
[the remark this answers is not in the document]
Comment (dasconnor) [reply pkg/prompts/prompts.go:30]: I believe this function performs that check

`func NewPrompter() Prompter {
	return &realPrompter{}
}`

File: pkg/vm/descriptor_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	// use io.Discard to not print anything
+	ux.NewUserLog(logging.NoLog{}, io.Discard)
+	return assert.New(t)
+}
+
+func Test_getChainId(t *testing.T) {
+	assert := setupTest(t)
+	app := application.New()
+	mockPrompt := &mocks.Prompter{}
+	app.Prompt = mockPrompt
```
[the remark this answers is not in the document]
Comment (dasconnor) [reply pkg/vm/descriptor_test.go:31]: actually it needs to be done this way to set the `mockPrompt.On` calls properly

File: pkg/vm/descriptor_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	token, err := getTokenName(app)
+	assert.NoError(err)
+	assert.Equal(testToken, token)
+}
+
+func Test_getChainId_Err(t *testing.T) {
+	assert := setupTest(t)
+	app := application.New()
+	mockPrompt := &mocks.Prompter{}
+	app.Prompt = mockPrompt
```
[the remark this answers is not in the document]
Comment (dasconnor) [reply pkg/vm/descriptor_test.go:44]: same as above

File: internal/mocks/getGRPCClientFunc.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
+// Code generated by mockery v2.14.0. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (dasconnor) [reply internal/mocks/getGRPCClientFunc.go:1]: I don't think so, but I'm ok with the extra generated files. We should probably add tests for all these eventually

File: pkg/vm/descriptor_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/vm/descriptor_test.go:31]: can't use app.Prompt.On ? whichever is more clear for you, in any case

File: scripts/regenerate_mocks.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if ! [[ "$0" =~ scripts/regenerate_mocks.sh ]]; then
+  echo "must be run from repository root"
+  exit 1
+fi
+
+go install github.com/vektra/mockery/v2@latest
+
+mockery --all --output ./internal/mocks
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply scripts/regenerate_mocks.sh]: I see. I was confused by _test.go (only spawn_server was snake and not test). Not good convention anyway.

Ok maybe we can add mock generation to ANR script for that. Creating in ANR repo a mock file that will be used by CLI? In that case, not copy wil be needed either.

File: pkg/vm/descriptor_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dasconnor) [reply pkg/vm/descriptor_test.go:31]: no, that is of type `prompt.Prompter`, not `mock.Prompter`. Only mock.Prompter has an `On` method.

File: internal/mocks/getGRPCClientFunc.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (dasconnor) [reply internal/mocks/getGRPCClientFunc.go]: I removed the extras

File: internal/mocks/setDefaultSnapshotFunc.go (hunk 1 above)
Comment (dasconnor) [reply internal/mocks/setDefaultSnapshotFunc.go]: removed

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**